js/modules/tamagotchi.js

```diff
@@ -37,4 +38,66 @@ export default class Tamagotchi {
     this.fun.element = funElement;
     this.displayFun(funElement);
   };
+
+  checkState() {
+    if (this.health.value <= 0) {
+      this.state = "dead";
+    } else if (this.energy.value <= 6) {
+      this.state = "sleepy";
+    } else if (this.hunger.value <= 6) {
+      this.state = "hungry";
+    } else if (this.fun.value <= 6) {
+      this.state = "bored";
+    } else if (
+      this.hunger.value >= 7 &&
+      this.energy.value >= 7 &&
+      this.fun.value >= 7
+    ) {
+      this.state = "happy";
+    }
+  }
+
+  updateState() {
+    this.checkState();
+
+    const nimo = document.getElementById("nimo");
+    const stateText = document.getElementById("petState");
+
+    nimo.classList = "";
+
+    switch (this.state) {
+      case "dead":
```

Comment:
**nitpick:** Same here, regarding reusing states name. Try to refactor it so you could do:

```
switch (this.state) {
      case this.states.dead:
          ...
```